src/ILCompiler.Compiler/src/Compiler/TypeExtensions.cs

```diff
@@ -79,7 +79,7 @@ public static bool RequiresInstMethodTableArg(this MethodDesc method)
         /// </summary>
         public static bool AcquiresInstMethodTableFromThis(this MethodDesc method)
         {
-            return method.IsSharedByGenericInstantiations &&
+            return method.IsCanonicalMethod(CanonicalFormKind.Any) &&
```

Comment:
I disagree, I believe we should make the IsSharedByGenericInstantiations in the typesystem work correctly, instead of requiring folks to call IsCanonicalMethod with a specific parameter.

---
In reply to: [81459466](https://github.com/dotnet/corert/pull/1969#discussion_r81459466) [](ancestors = 81459466)